### ISSUE TITLE
Docs: Update the 'Major Release Steps' docs for what we did to release v6

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6761]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

We've done a few things to prep v6 that aren't in the docs yet. They should be documented.

At the very least:

* #6759
* #6760
* Writing & sharing out a blog post
* Creating a Discord channel

### Affected URL(s)

https://typescript-eslint.io/maintenance/releases#major-release-steps

**Comment by JoshuaKGoldberg** [comment]
I think we'll need to update the list of expected status checks for the PRs whenever we update a Node version, too.

![Screenshot of several Run Unit Tests (18, ...) status checks still waiting to be reported in GitHub](https://github.com/typescript-eslint/typescript-eslint/assets/3335181/e861414c-e466-42a8-b4f9-231e8e191ed6)

**Comment by bradzacher** [comment]
> I think we'll need to update the list of expected status checks for the PRs whenever we update a Node version, too.

I just did that yeah - that's not specifically a major release thing though - that's just a CI change step - but we happened to change the min AND max node versions in this major which meant that all the old required steps are now out-of-date.

I just updated it all.
Steps are:
1) https://github.com/typescript-eslint/typescript-eslint/settings/branches
2) edit the `main` branch protection
3) remove the now obsolete steps from "Status checks that are required."
4) add the new steps to "Status checks that are required."

NOTE: this sucks, the UI sucks, and it's a pain in the ass to do, sadly.
